### PR TITLE
[#6099] Remove low impact FxCop exclusions - Sample cases (1/2)

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses #6099
#minor

## Description
This PR removes some of the FxCop rule exclusions with low impact.

### Detailed Changes
- **CA1819 - Properties should not return arrays**
   - Updated the `Records` property type in the `FeedbackRecords` class from array to collection.
- **CA1068 - CancellationToken parameters must come last**
   - Updated the `RecognizeAsync` method in the `Recognizer` class reordering the parameters.
- **CA1064 - Exceptions should be public**
   - Move protected class `InvokeResponseException` from the `ActivityHandler` class to its own file making it public.
- **CA1052 - Static holder types should be Static or NotInheritable**
   - Marked the `BotAssert` class as static.
- **CA1801 - Review unused parameters**
   - Removed unused CancellationToken parameter from `SetPropertyValueAsync` method of `BotState` class.
- **CA1034 - Nested types should not be visible**
   - Moved nested `CachedBotState` class to its own file.
- **CA1716 - Identifiers should not match keywords**
   - Renamed _next_ parameter of the `OnTurnAsync` method of `IMiddleware` interface as _nextDelegate_.
- **CA1055 - Uri return values should not be strings** 
   - Updated the `GetAttachmentUri` method of the `AttachmentEx` class to return Uri instead of string.
 - **CA1822 - Mark members as static**
   - Removing this rule exclusion in `TelemetryLoggerMiddleware` class didn't require changes.
- **CA1715 // Identifiers should have correct prefix**
   - Renamed generic type _TypeT_ as _TType_ in `GetChannelData` method in the `ActivityEx` class.
- **VSTHRD110 - Observe result of async calls** 
   - Added a variable to observe the result of Task.Run in `RequestManager` class.
- **VSTHRD200 - Use "Async" suffix for async methods**
   - Renamed the `Invoke` method of the `TelemetrySaveBodyASPMiddleware` class as `InvokeAsync`.

## Testing
This image shows the tests passing locally after the changes.
image

### CODE REVIEW

File: libraries/Microsoft.Bot.Schema/ActivityEx.cs [hunk 1]
```diff
@@ -407,42 +407,38 @@ public Mention[] GetMentions()
...
         /// </summary>
-        /// <typeparam name="TypeT">The type of the object to return.</typeparam>
+        /// <typeparam name="TType">The type of the object to return.</typeparam>
         /// <returns>The strongly-typed object; or the type's default value, if the <see cref="ChannelData"/> is null.</returns>
         /// <seealso cref="ChannelData"/>
         /// <seealso cref="TryGetChannelData{TypeT}(out TypeT)"/>
-#pragma warning disable CA1715 // Identifiers should have correct prefix (we can't change it without breaking binary compatibility)
-        public TypeT GetChannelData<TypeT>()
-#pragma warning restore CA1715 // Identifiers should have correct prefix
+        public TType GetChannelData<TType>()
```
Comment (gabog) [inline libraries/Microsoft.Bot.Schema/ActivityEx.cs:414]: Can we just use T instead of TType?

File: libraries/Microsoft.Bot.Schema/ActivityEx.cs [hunk 2]
```diff
@@ -407,42 +407,38 @@ public Mention[] GetMentions()
...
         /// <returns>
         /// <c>true</c> if the operation succeeded; otherwise, <c>false</c>.
         /// </returns>
         /// <seealso cref="ChannelData"/>
-        /// <seealso cref="GetChannelData{TypeT}"/>
-#pragma warning disable CA1715 // Identifiers should have correct prefix (we can't change it without breaking binary compatibility)
-        public bool TryGetChannelData<TypeT>(out TypeT instance)
-#pragma warning restore CA1715 // Identifiers should have correct prefix
+        /// <seealso cref="GetChannelData{TType}"/>
+        public bool TryGetChannelData<TType>(out TType instance)
```
Comment (gabog) [inline libraries/Microsoft.Bot.Schema/ActivityEx.cs:441]: Can we just use T instead of TType?

**Review by gabog — CHANGES_REQUESTED** [review]
See small comments below and let me know if they make sense.

Thanks

File: libraries/Microsoft.Bot.Schema/ActivityEx.cs (hunk 1 above)
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Schema/ActivityEx.cs]: Done. Thanks!

File: libraries/Microsoft.Bot.Schema/ActivityEx.cs (hunk 2 above)
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Schema/ActivityEx.cs]: Done. Thanks!

File: libraries/Microsoft.Bot.Schema/ActivityEx.cs [hunk 3]
```diff
@@ -1,5 +1,4 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
+﻿// Licensed under the MIT License.
```
Comment (gabog) [inline libraries/Microsoft.Bot.Schema/ActivityEx.cs:1]: (C) is incorrect
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
```

**Review by gabog — APPROVED** [review]

**Pull request merged**